packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -299,6 +300,10 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
             }
         });
 
+        commands.registerCommand({ id: 'workbench.action.debug.restart' }, {
```

Comment:
hello, why do we need to register it again ? 
it's not already registered there https://github.com/theia-ide/theia/blob/06d69ece396afa82401975db310ba074a9c2b36e/packages/debug/src/browser/debug-frontend-application-contribution.ts#L525 ?

Reply:
I have the only one possible answer... 
Yes, after I changed the `id` of `DebugCommands.RESTART` to `workbench.action.debug.restart` (previously it was `debug.restart`) the command and its handler got registered in `debug-frontend-application-contribution`. But it didn't help the `vscode-java-debug` extension to get the right command from the `CommandRegistry`, probably because the command returned into `vscode-java-debug`'s `hcrEvent` handler from the `CommandRegistry` by `id` cannot be cast to 
```
export interface ICommand {
	id: string;
	handler: ICommandHandler;
	description?: ICommandHandlerDescription | null;
}
```

without this code lines added to `plugin-vscode-commands-contribution`:
```
        commands.registerCommand({ id: 'workbench.action.debug.restart' }, {
            execute: () => commands.executeCommand(DebugSessionContextCommands.RESTART.id)
        });
```
HotCodeReplace was resulting into:
```
root ERROR [hosted-plugin: 24978] (node:24978) UnhandledPromiseRejectionWarning: Error: The command 'workbench.action.debug.restart' cannot be executed. There are no active handlers available for the command.
    at CommandRegistry.<anonymous> (http://localhost:3000/bundle.js:98782:31)
    at step (http://localhost:3000/bundle.js:98570:23)
    at Object.next (http://localhost:3000/bundle.js:98551:53)
    at http://localhost:3000/bundle.js:98545:71
    at new Promise (<anonymous>)
    at ../../packages/core/lib/common/command.js.__awaiter (http://localhost:3000/bundle.js:98541:12)
    at CommandRegistry.../../packages/core/lib/common/command.js.CommandRegistry.executeCommand (http://localhost:3000/bundle.js:98765:16)
    at CommandRegistryMainImpl.push.../../packages/plugin-ext/lib/main/browser/command-registry-main.js.CommandRegistryMainImpl.$executeCommand (http://localhost:3000/86.bundle.js:3301:72)
    at RPCProtocolImpl.push.../../packages/plugin-ext/lib/api/rpc-protocol.js.RPCProtocolImpl.doInvokeHandler (http://localhost:3000/44.bundle.js:498:23)
    at RPCProtocolImpl.push.../../packages/plugin-ext/lib/api/rpc-protocol.js.RPCProtocolImpl.invokeHandler (http://localhost:3000/44.bundle.js:483:41)

root ERROR [hosted-plugin: 24978] (node:24978) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 7)
```
I suppose that with this PR we actually have two different objects registered in `CommandRegistry` for the `workbench.action.debug.restart`-command. Or, if `id` is the key, than one of the objects is replaced by the other one. Not sure if it's good.

Or... Those `CommandRegistry` in `debug-frontend-application-contribution` and in `plugin-vscode-commands-contribution` are just two different objects...

Reply:
I've rolled back the change in `plugin-vscode-commands-contribution` that double-registers the command `workbench.action.debug.restart` - and the debug session restart still works when invoked from HCR event handler. Verified this with `vscode-java-debug` v.0.16.0 and v.0.18.0. 

So, I'm confused and can't find the reason for `root ERROR [hosted-plugin: 24978] (node:24978) UnhandledPromiseRejectionWarning: Error: The command 'workbench.action.debug.restart' cannot be executed. There are no active handlers available for the command.` that was appearing with `vscode-java-debug` v.0.16.0 (see:  https://github.com/theia-ide/theia/pull/5366#discussion_r290890804)

I've removed the exceeding command registration from the PR.